app/concerns/web_request_concern.rb

```diff
@@ -53,7 +53,7 @@ def call(env)
           # Return body as binary if default_encoding is nil
           next if encoding.nil?
         end
-        body.encode!(Encoding::UTF_8, encoding)
+        body.encode!(Encoding::UTF_8, encoding, invalid: :replace, undef: :replace, replace: " " )
```

Comment:
You can leave out the replace character (default: U+FFFD) so the user will know that there was an invalid character.
```suggestion
        body.encode!(Encoding::UTF_8, encoding, invalid: :replace, undef: :replace)
```